jsonschema/validators.py

```diff
@@ -135,7 +135,15 @@ def validate(self, *args, **kwargs):
                 raise error
 
         def is_type(self, instance, type):
-            if type not in self._types:
+            if type == u"object":
```

Comment:
Perhaps we should check for `object` in `default_types` first, and fall back to this?

Reply:
So, I'd rather widen the interface entirely here, rather than add complexity to the implementation.

In draft 6 for example, for some reason we decided to make `1.0` be an integer now, so doing `isinstance(float)` is no longer a reasonable implementation to check types.

That use case, combined with yours, makes me think that what we'll need to do now is to actually move away from the idea that types are a mapping between names and classes to check via isinstance, and that we'll have to just have e.g. `Validator(is_type=SomeInjectableFunction)` where the function will be passed 2 arguments, the same 2 as this is_type method. In your case for namedtuple, you'd inject in a function that first checks if it has your `_asdict` attribute and then calls up to the default implementation.

Does that sound reasonable?

Reply:
Makes sense. I'll see what I can do.

Reply:
I'm not sure you're still interested in this, but if you are, @bsmithers is also poking about in this area in a branch of his. I think he's made some progress on the track above -- if you're interested might want to sync up with him.

Gonna close this PR in the hope that we'll make enough progress there to resolve #364 but yeah this was super appreciated so definitely wouldn't mind some more hands if you are still pitching in.